src/communication/cmclient.cpp

```diff
@@ -74,6 +78,80 @@ aos::Error CMClient::ProcessMessage(const aos::String& methodName, uint64_t requ
  * Private
  **********************************************************************************************************************/
 
+aos::Error CMClient::ProcessGetUnitConfigStatus()
+{
+    LOG_DBG() << "Receive SM message: message = GetUnitConfigStatus";
+
+    auto  outgoingMessage = aos::MakeUnique<servicemanager_v3_SMOutgoingMessages>(&mAllocator);
+    auto& pbConfigStatus  = outgoingMessage->SMOutgoingMessage.unit_config_status;
+
+    outgoingMessage->which_SMOutgoingMessage = servicemanager_v3_SMOutgoingMessages_unit_config_status_tag;
+    pbConfigStatus                           = servicemanager_v3_UnitConfigStatus_init_zero;
+
+    aos::StaticString<aos::cVendorVersionLen> version;
+
+    auto err = mResourceManager->GetUnitConfigInfo(version);
+    if (!err.IsNone()) {
+        ErrorToPB(err, pbConfigStatus.error);
+    }
+
+    StringToPB(version, pbConfigStatus.vendor_version);
+
+    LOG_DBG() << "Send SM message: message = UnitConfigStatus";
+
+    return SendOutgoingMessage(*outgoingMessage);
+}
+
+aos::Error CMClient::ProcessCheckUnitConfig(const servicemanager_v3_CheckUnitConfig& pbUnitConfig)
+{
+    LOG_DBG() << "Receive SM message: message = CheckUnitConfig";
+
+    auto  outgoingMessage = aos::MakeUnique<servicemanager_v3_SMOutgoingMessages>(&mAllocator);
+    auto& pbConfigStatus  = outgoingMessage->SMOutgoingMessage.unit_config_status;
+
+    outgoingMessage->which_SMOutgoingMessage = servicemanager_v3_SMOutgoingMessages_unit_config_status_tag;
+    pbConfigStatus                           = servicemanager_v3_UnitConfigStatus_init_zero;
+
+    auto err = mResourceManager->CheckUnitConfig(pbUnitConfig.vendor_version, pbUnitConfig.unit_config);
+    if (!err.IsNone()) {
+        ErrorToPB(err, pbConfigStatus.error);
+    }
+
+    aos::StaticString<aos::cVendorVersionLen> version;
+
+    PBToString(pbUnitConfig.vendor_version, version);
+    StringToPB(version, pbConfigStatus.vendor_version);
```

Comment:
I think it should be possible to copy pbUnitConfig.vendor_version into pbConfigStatus.vendor_version without additional manipulations.